data/modules/SearchRescue/SearchRescue.lua

```diff
@@ -661,7 +668,10 @@ local createTargetShipParameters = function (flavour)
 	if flavour.id == 1 or flavour.id == 6 then
 		local any_pass = rand:Integer(0,1)
 		if any_pass > 0 then
-			pickup_pass = rand:Integer(1, math.min((shipdef.capacity / 10)+1, max_pass))
+			local drive_mass = 4
```

Comment:
same as above. (do you need to set it twice?)

Reply:
I need to do it twice. First time shipdefs are filtered based on projected need for equipment capacity and several other things (depending on mission specifics). Second time the actual amount of passengers to be rescued is calculated and fed back to the mission. Both have to happen before the ad is created.